### ISSUE TITLE
Numerically Stable Softmax

### ISSUE THREAD

**Issue by kellywzhang** [kellywzhang/reading-comprehension#1]
Need to change custom softmax to make it numerically stable (exponentials can explode) in attention.py and StanfordReader.py

**Comment by kellywzhang** [comment]
See here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/api_docs/python/functions_and_classes/shard5/tf.nn.sigmoid_cross_entropy_with_logits.md

**Comment by kellywzhang** [comment]
Even better:
http://deeplearning.net/software/theano/library/tensor/nnet/nnet.html#theano.tensor.nnet.nnet.softmax

**Comment by kellywzhang** [comment]
Implemented in attention.py and StanfordReader.py
